### PR TITLE
Feature/appauth test

### PR DESCRIPTION
**Pull request by charlon**

### CODE REVIEW

File: myuw-ios/AppAuthTest.swift [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+//  Created by Charlon Palacay on 1/7/20.
+//  Copyright © 2020 Charlon Palacay. All rights reserved.
+//
+
+import AppAuth
+import UIKit
+
+typealias PostRegistrationCallback = (_ configuration: OIDServiceConfiguration?, _ registrationResponse: OIDRegistrationResponse?) -> Void
+
+let kIssuer: String = "https://accounts.google.com";
```
Comment (abztrakt) [inline myuw-ios/AppAuthTest.swift]: This should probably be in the plist.

File: myuw-ios/AppAuthTest.swift [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+let kClientID: String? = clientID
+// can also use reverse DNS notion of the client ID for kRedirectURI
+let kRedirectURI: String = "edu.uw.myuw-ios:/";
```
Comment (abztrakt) [inline myuw-ios/AppAuthTest.swift:18]: This should be in the plist too - we may have reason to set it to the one provided by the idp in the future.

File: myuw-ios/AppAuthTest.swift [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+let kAppAuthExampleAuthStateKey: String = "authState";
```
Comment (abztrakt) [inline myuw-ios/AppAuthTest.swift:19]: Example?

File: myuw-ios/AppAuthTest.swift [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+                    self.doAuthWithAutoCodeExchange(configuration: configuration,
+                                                    clientID: clientID,
+                                                    clientSecret: response?.clientSecret)
+                }
+            
+            }
+        }
+    }
+    
+    func authNoCodeExchange() {
```
Comment (abztrakt) [inline myuw-ios/AppAuthTest.swift:106]: So, in the long run, is there a reason to implement both this and authWithAutoCodeExchange?

File: myuw-ios/AppAuthTest.swift [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+                                              responseType: OIDResponseTypeCode,
+                                              additionalParameters: nil)
+
+        // performs authentication request
+        print("Initiating authorization request with scope: \(request.scope ?? "DEFAULT_SCOPE")")
+
+        appDelegate.currentAuthorizationFlow = OIDAuthState.authState(byPresenting: request, presenting: self) { authState, error in
+            if let authState = authState {
+                self.setAuthState(authState)
+                print("Got authorization tokens. Access token: \(authState.lastTokenResponse?.accessToken ?? "DEFAULT_TOKEN")")
```
Comment (abztrakt) [inline myuw-ios/AppAuthTest.swift]: Rather than print, we should start logging these with an appropriate loglevel.

**Review by abztrakt — CHANGES_REQUESTED** [review]

File: myuw-ios/AppAuthTest.swift (hunk 5 above)
Comment (charlon) [reply myuw-ios/AppAuthTest.swift]: using os_log now.

File: myuw-ios/AppAuthTest.swift (hunk 4 above)
Comment (charlon) [reply myuw-ios/AppAuthTest.swift:106]: i think we'll want to keep the AutoCodeExchange stuff for PKCE.

**Pull request merged**